### ISSUE TITLE
static build still broken

### ISSUE THREAD

**Issue by humeafo** [stp/stp#219]
it seems that staticly linked the pthread library and results:
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../x86_64-linux-gnu/libpthread.a(nptl-init.o): In function `__pthread_initialize_minimal_internal':
/build/buildd/eglibc-2.19/nptl/nptl-init.c:285: undefined reference to `__libc_setup_tls'
/build/buildd/eglibc-2.19/nptl/nptl-init.c:303: undefined reference to `_dl_cpuclock_offset'
/build/buildd/eglibc-2.19/nptl/nptl-init.c:419: undefined reference to `_dl_pagesize'
/build/buildd/eglibc-2.19/nptl/nptl-init.c:428: undefined reference to `_dl_pagesize'
/build/buildd/eglibc-2.19/nptl/nptl-init.c:448: undefined reference to `_dl_init_static_tls'

**Comment by humeafo** [comment]
my system is ubuntu 14.04

**Comment by msoos** [comment]
Well, let's start with the basics. I have ubuntu 14.04 so it should work :) First of all, please delete the "build" directory, then re-create it, and then:

```
cmake -DENABLE_TESTING=ON -DBUILD_STATIC_BIN=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DFORCE_CMS=ON   ..
make -j4 VERBOSE=1
```

And please give me the output of ALL of that, *everything*. So we can see what's going on. Also, please remove CryptoMiniSat (in case you have it installed). We can then debug :)

**Comment by msoos** [comment]
OOOps, sorry, I had to edit the command above, please use the edited command :)

**Comment by humeafo** [comment]
remove CryptoMiniSat  and use your commands worked:
cmake -DBUILD_STATIC_BIN=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF   ..
make -j4 VERBOSE=1

but 
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF   ..
make -j4 VERBOSE=1

still not work, I thought BUILD_SHARED_LIBS implies that BUILD_STATIC_BIN ON, but obviously this is not the case...

**Comment by humeafo** [comment]
now I found the problem, when added CM back using build command:
cmake .. -DALSO_BUILD_STATIC_LIB=ON -DNOM4RI=ON

no problem, it seemed m4ri dynamic linked caused the failure...

BTW, I tried to built CM with:
cmake ..  -DNOM4RI=ON -DSTATICCOMPILE=ON

a bunch of errors occurred...maybe this should be discussed in CM issue list?

**Comment by msoos** [comment]
I tested, and this:

```
cmake -DENABLE_TESTING=ON -DREQUIRE_M4RI=ON -DSTATICCOMPILE=ON -DENABLE_PYTHON_INTERFACE=OFF -DNOVALGRIND=ON -DNOZLIB=ON -DNOM4RI=ON -DONLY_SIMPLE=ON -DSTATICCOMPILE=ON -DCMAKE_BUILD_TYPE=Release ..
make
```

works for cryptominisat. It doesn't build the cryptominisat4 binary, though, only the cryptominisat4_simple. I plan to remove the _simple though and get rid of the boost dependency. Sorry about it being broken in the meanwhile. The library does get built with _simple, though and it can be used in STP in full.

I gather that all the rest has been fixed :) The problem is that there are way too many build options and it's hard to support them all. I'm working on cleaning some things up :)
